mods/dungeon_loot/mapgen.lua

```diff
@@ -0,0 +1,159 @@
+minetest.set_gen_notify({dungeon=true})
+
+local function noise3d_integer(noise, pos)
+	return math.abs(math.floor(noise:get3d(pos) * 2147483647))
+end
+
+local function random_sample(rand, list, count)
+	local ret = {}
+	for n = 1, count do
+		local idx = rand:next(1, #list)
+		table.insert(ret, list[idx])
+		table.remove(list, idx)
+	end
+	return ret
+end
+
+local function find_walls(cpos)
+	local wall = ItemStack("mapgen_cobble"):get_name()
```

Comment:
does this have any benefits over the following? Seems a bit hacky to me.

```lua
minetest.registered_aliases["mapgen_cobble"]
```

Reply:
any benefits? no
i wouldn't call this hacky tho, just <i>clever</i>